### PR TITLE
Support if-statement in CuPy JIT

### PR DESCRIPTION
**Pull request by asi1024**
Blocked by #4633.

### CODE REVIEW

File: cupyx/jit/_compile.py [hunk 1]
```diff
@@ -285,15 +297,22 @@ def _transpile_stmt(stmt, env):
...
         raise NotImplementedError('Not implemented.')
     if isinstance(stmt, ast.If):
-        raise NotImplementedError('Not implemented.')
+        condition = _transpile_expr(stmt.test, env)
+        if is_constants([condition]):
+            stmts = stmt.body if condition.obj else stmt.orelse
+            return _transpile_stmts(stmts, env)
+        head = f'if ({condition.code})'
+        then_body = _transpile_stmts(stmt.body, env)
+        else_body = _transpile_stmts(stmt.orelse, env)
```
Comment (leofang) [inline cupyx/jit/_compile.py:314]: How about an `if ... elif ... elif ... else` block? Does `elif` get handled here?

By the way, I noticed Cython translates such a block to a C switch when the condition is simple enough (ex: condition is a runtime constant). Do you think it's possible to do so here? Just curious 🙂

Comment (asi1024) [reply cupyx/jit/_compile.py:314]: Python's `ast` module handles `if ... elif ... else ...` equivalent to `if ... else { if ... else ... }`.
For example, in the test case [test_elif](https://github.com/cupy/cupy/pull/4646/files#diff-e0187cde046aeb239283aa518edbbac8aae3c09193ab03ccc82d2c607e79f1f3R324-R332), cupy.vectorize generates the following cuda code.
```cpp
template <typename T0, typename T1>
__device__ long long cupy_remainder_int64(T0 in0, T1 in1) {
    return (in0 - _floor_divide(in0, in1) * in1) * (in1 != 0);
}

__device__ long long func_if(long long x) {
  long long y;
  y = 0ll;
  if ((cupy_remainder_int64(x, 2ll) == 0ll)) {
    y = x;
  }
  else {
    if ((cupy_remainder_int64(x, 3ll) == 0ll)) {
      y = (-x);
    }
    else {
    }
  }
  return y;
}
```

**Comment by asi1024** [comment]
Rebased!

File: cupyx/jit/_typerules.py [hunk 2]
```diff
@@ -132,4 +136,29 @@ def get_ctype_from_scalar(mode, x):
...
             return _types.Scalar(numpy.complex64)
 
-    raise NotImplementedError(f'{x} is not supported as a constant.')
+    raise NotImplementedError(f'{x} is not scalar object.')
+
+
+_suffix_literals_dict = {
+    numpy.dtype('float64'): '',
+    numpy.dtype('float32'): 'f',
+    numpy.dtype('int64'): 'll',
```
Comment (emcastillo) [inline cupyx/jit/_typerules.py:145]: I am not sure, but could this cause problems in windows?
I think it should be ok, but double check is not bad. @leofang

Comment (leofang) [reply cupyx/jit/_typerules.py:145]: Prefer q over l whenever possible.

File: cupyx/jit/_compile.py (hunk 1 above)
Comment (leofang) [reply cupyx/jit/_compile.py:314]: Cool, didn't know the ast's behavior. Thanks!

**Review by emcastillo — COMMENTED** [review]
LGTM!!
The only thing is the dtype concern

File: cupyx/jit/_typerules.py (hunk 2 above)
Comment (asi1024) [reply cupyx/jit/_typerules.py:145]: Fixed in b84eb5f!

**Comment by emcastillo** [comment]
Jenkins, test this please.

Awesome work!

**Review by emcastillo — APPROVED** [review]

**Pull request merged**